app/models/spree/gateway/amazon.rb

```diff
@@ -70,12 +70,18 @@ def authorize(amount, amazon_checkout, gateway_options={})
       if amount < 0
         return ActiveMerchant::Billing::Response.new(true, "Success", {})
       end
-      order = Spree::Order.find_by(:number => gateway_options[:order_id].split("-")[0])
+      order_number, payment_number = gateway_options[:order_id].split("-", 2)
+      order = Spree::Order.find_by!(number: order_number)
+      payment = Spree::Payment.find_by!(number: payment_number)
+      authorization_reference_id = "#{payment.number}-#{random_suffix}"
```

Comment:
Maybe we can extract a method called `operation_uid(payment)` to abstract generating the identifier and use that on `authorize`, `capture` and `refund` 

```ruby
def operation_uid(payment)
  "#{payment.number}-#{random_suffix}"
end
```